cql3/restrictions/statement_restrictions.cc

```diff
@@ -1021,5 +1022,42 @@ ::shared_ptr<single_column_restriction> single_column_restriction::LIKE::apply_t
     return r;
 }
 
+namespace wip {
+
+::shared_ptr<expression> make_conjunction(::shared_ptr<expression> a, ::shared_ptr<expression> b) {
+    if (!a) {
+        return b;
+    }
+    if (!b) {
+        return a;
+    }
+    return std::visit(overloaded_functor{
+            [&] (const conjunction& a_conj) {
+                auto operands = a_conj.children;
+                std::visit(overloaded_functor{
+                        [&] (const conjunction& b_conj) {
+                            operands.insert(operands.end(), b_conj.children.cbegin(), b_conj.children.cend());
+                        },
+                        [&] (const binary_operator&) {
+                            operands.push_back(b);
+                        },
+                    }, *b);
+                return ::make_shared<expression>(conjunction{operands});
+            },
+            [&] (const binary_operator& a_binop) {
+                return std::visit(overloaded_functor{
+                        [&] (const conjunction&) {
+                            return make_conjunction(b, a);
+                        },
+                        [&] (const binary_operator&) {
+                            return ::make_shared<expression>(conjunction{std::vector{a, b}});
+                        },
+                    }, *b);
+            },
+        }, *a);
```

Comment:
Here's a nicer (IMO) way to code this:

```c++
std::vector<shared_ptr<expression>> explode_conjunction(shared_ptr<expression> expr) {
    // returns  expr->children if it is a conjuction, otherwise {expr}, and maybe {} if expr is null
}

shared_ptr<expression> make_conjunction(shared_ptr<expression> a, shared_ptr<expression> b) {
    auto u = explode_conjunction(a);
    auto v = explode_conjunction(b);
    u.insert(e.end(), v.begin(), v.end());
    return make_shared<expression>(conjunction(std::move(u)));
}
```

Reply:
Love it!  Done.